physionet-django/waveforms/dash_apps/finished_apps/waveform_vis.py

```diff
@@ -0,0 +1,801 @@
+    base_fig.update_layout({
+        'height': max_plot_height / 2,
+        'width': fig_width,
+        'margin': {
+            'l': margin_left,
+            't': margin_top,
+            'r': margin_right,
+            'b': margin_bottom
+        },
+        'grid': {
+            'rows': 1,
+            'columns': 1,
+            'pattern': 'independent'
+        },
+        'showlegend': False,
+        'hovermode': 'x',
+        'dragmode': drag_mode
+    })
+    # Update the Null signal and axes
+    base_fig.add_trace(go.Scatter({
+        'x': [None],
+        'y': [None]
+    }), row = 1, col = 1)
+    # Update axes based on signal type
+    x_tick_vals = [round(n,1) for n in np.arange(0, 10.1, grid_delta_major).tolist()]
+    x_tick_text = [str(round(n)) if n%1 == 0 else '' for n in x_tick_vals]
+    y_tick_vals = [round(n,1) for n in np.arange(0, 2.25, grid_delta_major).tolist()]
+    y_tick_text = [str(n) if n%1 == 0 else ' ' for n in y_tick_vals]
+    # Create the empty chart
+    base_fig.update_xaxes({
+        'title': 'Time (s)',
+        'fixedrange': x_zoom_fixed,
+        'showgrid': True,
+        'tickvals': x_tick_vals,
+        'ticktext': x_tick_text,
+        'showticklabels': True,
+        'gridcolor': gridzero_color,
+        'zeroline': False,
+        'zerolinewidth': 1,
+        'zerolinecolor': gridzero_color,
+        'gridwidth': 1,
+        'range': [0, 10],
+        'rangeslider': {
+            'visible': False
+        }
+    }, row = 1, col = 1)
+    base_fig.update_yaxes({
+        'fixedrange': y_zoom_fixed,
+        'showgrid': True,
+        'dtick': None,
+        'showticklabels': True,
+        'gridcolor': gridzero_color,
+        'zeroline': False,
+        'gridwidth': 1,
+        'range': [0, 2.25],
+    }, row = 1, col = 1)
+    return (base_fig)
+
+def get_xaxis(title, x_zoom_fixed, tick_labels, tick_vals, tick_text,
+              gridzero_color, start_time, range_stop):
+    return {
+        'title': title,
+        'fixedrange': x_zoom_fixed,
+        'dtick': 0.2,
+        'showticklabels': tick_labels,
+        'tickvals': tick_vals,
+        'ticktext': tick_text,
+        'gridcolor': gridzero_color,
+        'gridwidth': 1,
+        'zeroline': False,
+        'range': [start_time, range_stop],
+        'showspikes': True,
+        'spikemode': 'across',
+        'spikesnap': 'cursor',
+        'showline': True
+    }
+
+def window_signal(y_vals):
+    """
+    This uses the Coefficient of Variation (CV) approach to determine
+    significant changes in the signal then return the adjusted minimum
+    and maximum range... If a significant variation is signal is found
+    then filter out extrema using normal distribution
+    """
+    # Standard deviation signal range to window
+    std_range = 4
+    # Get parameters of the signal
+    temp_std = stats.median_absolute_deviation(y_vals, nan_policy='omit')
+    temp_std = np.nanstd(y_vals)
```

Comment:
You assign `temp_std` to the median absolute deviation (MAD) and then overwrite it with standard deviation.  If you don't want to use MAD any longer, you can remove this line.  If you do want to use MAD, you may consider revising the comment above since Coefficient of Variation implies using standard deviation.

Reply:
Good catch! I thought I switched not sure how it got back... will fix!